### PR TITLE
completing the switch to github pages for docs

### PR DESCRIPTION
**Pull request by 2bndy5**
- Update the docs badge in the root README file.
- Modify the CI workflow to properly deploy to the gh-pages branch

### CODE REVIEW

File: .github/workflows/mkdocs-deploy.yml [hunk 1]
```diff
@@ -15,8 +13,11 @@ jobs:
       - uses: actions/setup-python@v2
         with:
           python-version: 3.x
-      - run: pip install -r docs/requirements.txt
-      - run: |
+      - name: Install python action for doc extraction
+        run: |
+          pip install . -r docs/requirements.txt
```
Comment (2bndy5) [inline .github/workflows/mkdocs-deploy.yml:18]: This should be a 1 line cmd
```suggestion
        run: pip install . -r docs/requirements.txt
```

**Comment by 2bndy5** [comment]
I'd prefer this get squashed when merged because the git history is a bit laborious.

**Comment by 2bndy5** [comment]
@shenxianpeng If you want, I can delete the docs hosted on RTD. They're duplicates now anyway.

**Comment by shenxianpeng** [comment]
> @shenxianpeng If you want, I can delete the docs hosted on RTD. They're duplicates now anyway.

Sure, it's OK for me.

**Review by shenxianpeng — APPROVED** [review]

**Comment by 2bndy5** [comment]
Done. The cpp-linter-action project no longer exists on RTD.

**Pull request merged**